ironfish/src/rpc/routes/accounts/getNotes.ts

```diff
@@ -3,59 +3,63 @@
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
 import * as yup from 'yup'
 import { ApiNamespace, router } from '../router'
-import { getAccount, getTransactionNotes } from './utils'
+import { getAccount } from './utils'
 
-export type GetAccountNotesRequest = { account?: string }
+export type GetAccountNotesStreamRequest = { account?: string }
 
-export type GetAccountNotesResponse = {
+export type GetAccountNotesStreamResponse = {
   account: string
-  notes: {
+  note: {
     owner: boolean
     amount: number
     memo: string
     transactionHash: string
     spent: boolean | undefined
-  }[]
+  }
 }
 
-export const GetAccountNotesRequestSchema: yup.ObjectSchema<GetAccountNotesRequest> = yup
-  .object({
-    account: yup.string().strip(true),
-  })
-  .defined()
+export const GetAccountNotesStreamRequestSchema: yup.ObjectSchema<GetAccountNotesStreamRequest> =
+  yup
+    .object({
+      account: yup.string().strip(true),
+    })
+    .defined()
 
-export const GetAccountNotesResponseSchema: yup.ObjectSchema<GetAccountNotesResponse> = yup
-  .object({
-    account: yup.string().defined(),
-    notes: yup
-      .array(
-        yup
-          .object({
-            owner: yup.boolean().defined(),
-            amount: yup.number().defined(),
-            memo: yup.string().trim().defined(),
-            transactionHash: yup.string().defined(),
-            spent: yup.boolean(),
-          })
-          .defined(),
-      )
-      .defined(),
-  })
-  .defined()
+export const GetAccountNotesStreamResponseSchema: yup.ObjectSchema<GetAccountNotesStreamResponse> =
+  yup
+    .object({
+      account: yup.string().defined(),
+      note: yup
+        .object({
+          owner: yup.boolean().defined(),
+          amount: yup.number().defined(),
+          memo: yup.string().trim().defined(),
+          transactionHash: yup.string().defined(),
+          spent: yup.boolean(),
+        })
+        .defined(),
+    })
+    .defined()
 
-router.register<typeof GetAccountNotesRequestSchema, GetAccountNotesResponse>(
-  `${ApiNamespace.account}/getAccountNotes`,
-  GetAccountNotesRequestSchema,
+router.register<typeof GetAccountNotesStreamRequestSchema, GetAccountNotesStreamResponse>(
+  `${ApiNamespace.account}/getAccountNotesStream`,
+  GetAccountNotesStreamRequestSchema,
   async (request, node): Promise<void> => {
     const account = getAccount(node, request.data.account)
 
-    const responseNotes = []
-
-    for await (const { transaction } of account.getTransactions()) {
-      const notes = await getTransactionNotes(account, transaction)
-      responseNotes.push(...notes)
+    for await (const { note, spent, transactionHash } of account.getNotes()) {
+      request.stream({
+        account: account.displayName,
```

Comment:
I think we should consider getting rid of this now. I was gonna remove it in the transaction stream.

Reply:
What about the case where we call getNotes with no account and return the notes from the default account. Wouldn't we want to return the name of the default account in this case?